### PR TITLE
Add empty pinned items section banner & new empty state svg

### PR DESCRIPTION
**Pull request by daltojohnso**
Related to #19758.

Adds banner + new empty state svg when there are no collection items. Not making the banner removable yet since there's no BE support for it.

Screen Shot 2022-01-22 at 1 56 05 PM
Screen Shot 2022-01-22 at 1 56 11 PM

### CODE REVIEW

**Review by mazameli — CHANGES_REQUESTED** [review]
The distance between the banner and the empty state content should ideally be a consistent ~40px to prevent awkward gaps like this (i.e., the placeholder images should align to the top I think): 

image


Also, in an ideal world I think this content would have a min-width, and then stretch wider to better fill up wide screens; but only if that's trivially easy.

image

**Comment by daltojohnso** [comment]
@mazameli I think getting rid of the defined `width` and `height` of the SVG gets us most of the way there; how is that?

**Comment by mazameli** [comment]
@daltojohnso looking now

**Review by mazameli — APPROVED** [review]
I pushed a commit with a reasonable maxWidth for the whole thing, and I think this is a good-enough happy medium for our purposes now.

File: frontend/src/metabase/collections/components/EmptyPinnedItemsBanner/EmptyPinnedItemsBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { t } from "ttag";
+
+import { EmptyBanner, ColoredIcon } from "./EmptyPinnedItemsBanner.styled";
+
+function EmptyPinnedItemsBanner() {
+  return (
+    <EmptyBanner className={undefined}>
+      <ColoredIcon name="pin" />
+      {t`Save your questions, dashboads, and datasets in collections — and pin them to feature them at the top.`}
```
Comment (kulyk) [inline frontend/src/metabase/collections/components/EmptyPinnedItemsBanner/EmptyPinnedItemsBanner.tsx:10]: Please replace "datasets" with "models"

File: frontend/src/metabase/collections/components/PinnedItemOverview/PinnedItemOverview.unit.spec.js [hunk 2]
```diff
@@ -45,8 +45,10 @@ function setup({ items, collection } = {}) {
 }
 
 describe("PinnedItemOverview", () => {
-  it("should render nothing if there are no items", () => {
+  it("should render an empty banner when there are no items", () => {
     const { container } = setup({ items: [] });
-    expect(container.firstChild).toBeNull();
+    expect(container.textContent).toContain(
+      "Save your questions, dashboads, and datasets in collections — and pin them to feature them at the top.",
```
Comment (kulyk) [inline frontend/src/metabase/collections/components/PinnedItemOverview/PinnedItemOverview.unit.spec.js:51]: Please replace "datasets" with "models"

**Pull request merged**